docs/xks/developer-guide/best-practices.md

```diff
@@ -3,18 +3,125 @@ id: best-practices
+2. You most likely want a basic readiness probe. This readiness probe should probably not involve other applications as this can lead to thundering herd problems. An endpoint that answers with **200** on **/healthz** on your http server for your application is a very good start.
+
+Your mileage may vary and only you can know what it means for your application to be ready to receive traffic. Consider reading this https://srcco.de/posts/kubernetes-liveness-probes-are-dangerous.html as it provides a more nuanced discussion when and where you want to use liveness/readiness probes.
+
+## Incoming HTTP
+
+TODO
+
+## Outgoing HTTP
+
+TODO
+
+## Database
+
+Your pod can be shut down by kubernetes. Make sure the following has been configured.
+
+1. Your application has graceful shutdown that closes connection to database.
+
+## Observability and Telemetry
+
+Make sure you have set up sufficient observability tools for your application. What this means depends on context. At Xenit we have found that the following provides a good start.
+
+### Logs
+
+1. Log the correct amount. This usually means one info log per "thing that happened". If you add too much you will never find the log you are looking for.
+2. Always add context to logs. A log should always be related to the entity that was affected. If you don't add this information the log is just noise and will never help you find issues in the future.
+3. Assume things work and mostly log errors. We have found that most info logs are just a poor man's implementation of metrics and traces. The ideal log is the log that immediately tells you what is wrong with the system. If you can get away with not logging something, consider doing so.
+4. Consider disabling HTTP request logging for non error http requests. We have found that it is common to add a general http log on top of every application. If you have a lot of traffic this will mean a lot of logs. How useful is it really to log that someone made a succesful _GET_ request to a certain endpoint?
+
+### Metrics
+
+Metrics is also a nuanced topic that depends on your context. However, it is usually a good idea to add the so called **RED** metrics to your application. **RED** stand for _Rate_, _Errors_, and _Duration_. We have found that measuring the rate of incoming requests, the rate of them erroring and the time it takes to execute them provides a very good baseline for knowing how your applications are doing.
+
+Consider reading our extended [extended documentation on metrics](https://xenitab.github.io/docs/xks/developer-guide/observability).
+
+### Traces
+
+Add tracing to your application. We have found that modern trace tools provide pretty good configuration out of the box. You just need to add an appropriate tracing library to your application. Consider reading our extended [extended documentation on tracing](https://xenitab.github.io/docs/xks/developer-guide/observability).
+
+## Pod disruption budgets
```

Comment:
Maybe change this to availability or something similar. I think this PDB documentation is to in depth for a overview page.

I think you should mention pdb as a part of getting good avaliability in k8s together with things like antiaffinity and link to our other docs for more details.